central/declarativeconfig/updater/group_updater.go

```diff
@@ -28,3 +40,27 @@ func (u *groupUpdater) Upsert(ctx context.Context, m proto.Message) error {
 	}
 	return u.groupDS.Upsert(ctx, group)
 }
+
+func (u *groupUpdater) DeleteResources(ctx context.Context, resourceIDsToSkip ...string) ([]string, error) {
+	resourcesToSkip := set.NewFrozenStringSet(resourceIDsToSkip...)
+
+	groups, err := u.groupDS.GetFiltered(ctx, func(group *storage.Group) bool {
+		return group.GetProps().GetTraits().GetOrigin() == storage.Traits_DECLARATIVE &&
+			!resourcesToSkip.Contains(group.GetProps().GetId())
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "retrieving declarative groups")
+	}
+
+	var groupDeletionErr *multierror.Error
+	var groupIDs []string
+	for _, group := range groups {
+		if err := u.groupDS.Remove(ctx, group.GetProps(), true); err != nil {
```

Comment:
I don't think this statement is true. If both the groups and auth providers are deleted, first the groups will be deleted, and afterwards the auth provider. So, this will never actually throw an error. And even if it was the "other way around", i.e. we have no specific order for deletion, since these are being done sequential, it would also never throw an error.

>but move them to an orphaned state as we talked about before.

Under what condition are we talking about now? Are we talking about ROX-14700 where we deal with "stale" references? Specifically, of a declarative group referencing a deleted declarative auth provider? Then this is in scope of this ticket for me, and should be dealt within the scope of that. One of the major challenges with groups is that there's no real workflow for users to discover groups other than through the auth provider page.

Reply:
> If both the groups and auth providers are deleted, first the groups will be deleted, and afterwards the auth provider.
Right, forgot about that.

I agree, let's deal with that in ROX-14700